src/coreComponents/physicsSolvers/multiphysics/PoroelasticSolverEmbeddedFractures.cpp

```diff
@@ -0,0 +1,864 @@
+#include "common/DataLayouts.hpp"
+#include "constitutive/ConstitutiveManager.hpp"
+#include "constitutive/contact/ContactRelationBase.hpp"
+#include "constitutive/solid/PoroElastic.hpp"
+#include "constitutive/fluid/SingleFluidBase.hpp"
+#include "managers/NumericalMethodsManager.hpp"
+#include "finiteElement/Kinematics.h"
+#include "linearAlgebra/solvers/BlockPreconditioner.hpp"
+#include "linearAlgebra/solvers/SeparateComponentPreconditioner.hpp"
+#include "managers/DomainPartition.hpp"
+#include "mesh/MeshForLoopInterface.hpp"
+#include "meshUtilities/ComputationalGeometry.hpp"
+#include "physicsSolvers/fluidFlow/SinglePhaseBase.hpp"
+#include "physicsSolvers/solidMechanics/SolidMechanicsLagrangianFEM.hpp"
+#include "physicsSolvers/solidMechanics/SolidMechanicsEmbeddedFractures.hpp"
+#include "rajaInterface/GEOS_RAJA_Interface.hpp"
+#include "linearAlgebra/interfaces/BlasLapackLA.hpp"
+#include "physicsSolvers/solidMechanics/SolidMechanicsEFEMKernelsHelper.hpp"
+
+namespace geosx
+{
+
+using namespace dataRepository;
+using namespace constitutive;
+
+PoroelasticSolverEmbeddedFractures::PoroelasticSolverEmbeddedFractures( const std::string & name,
+                                                                        Group * const parent ):
+  PoroelasticSolver( name, parent ),
+  m_fracturesSolverName()
+{
+  registerWrapper( viewKeyStruct::fracturesSolverNameString, &m_fracturesSolverName )->
+    setInputFlag( InputFlags::REQUIRED )->
+    setDescription( "Name of the fractures solver to use in the fractured poroelastic solver" );
+
+  this->getWrapper< string >( viewKeyStruct::discretizationString )->
+    setInputFlag( InputFlags::FALSE );
+}
+
+PoroelasticSolverEmbeddedFractures::~PoroelasticSolverEmbeddedFractures()
+{
+  // TODO Auto-generated destructor stub
+}
+
+void PoroelasticSolverEmbeddedFractures::postProcessInput()
+{
+  PoroelasticSolver::postProcessInput();
+
+  m_fracturesSolver  = this->getParent()->getGroup< SolidMechanicsEmbeddedFractures >
+                         ( m_fracturesSolverName );
+
+  GEOSX_ERROR_IF( m_fracturesSolver == nullptr,
+                  "Fractures solver not found or invalid type: " << m_fracturesSolverName );
+
+}
+
+void PoroelasticSolverEmbeddedFractures::registerDataOnMesh( dataRepository::Group * const meshBodies )
+{
+  PoroelasticSolver::registerDataOnMesh( meshBodies );
+
+  for( auto & mesh : meshBodies->getSubGroups() )
+  {
+    MeshLevel * meshLevel = Group::groupCast< MeshBody * >( mesh.second )->getMeshLevel( 0 );
+
+    ElementRegionManager * const elemManager = meshLevel->getElemManager();
+    {
+      elemManager->forElementRegions< SurfaceElementRegion >( [&] ( SurfaceElementRegion & region )
+      {
+        region.forElementSubRegions< EmbeddedSurfaceSubRegion >( [&]( EmbeddedSurfaceSubRegion & subRegion )
+        {
+          subRegion.registerWrapper< array1d< real64 > >( viewKeyStruct::dTraction_dPressureString );
+        } );
+      } );
+    }
```

Comment:
```suggestion
    elemManager->forElementRegions< SurfaceElementRegion >( [&] ( SurfaceElementRegion & region )
    {
      region.forElementSubRegions< EmbeddedSurfaceSubRegion >( [&]( EmbeddedSurfaceSubRegion & subRegion )
      {
        subRegion.registerWrapper< array1d< real64 > >( viewKeyStruct::dTraction_dPressureString );
      } );
    } );
```

Reply:
what s the correction here? indentation?

Reply:
yes